server/zanata-war/src/main/java/org/zanata/action/ProjectHome.java

```diff
@@ -688,6 +688,20 @@ public void initialize() {
         }
     }
 
+    public void onProjectNameChange(ValueChangeEvent e) {
+        if (!isValidName((String) e.getNewValue())) {
+            String componentId = e.getComponent().getId();
+            facesMessages.addToControl(componentId,
+                    "This project name is must be alphanumeric");
```

Comment:
"Project name must be alphanumeric" - also, probably should be a Messages object.